database/migrations/2020_05_01_121000_create_order_items_table.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use Tipoff\Checkout\Models\Order;
+use Tipoff\Checkout\Models\OrderItem;
+
+class CreateOrdersTable extends Migration
+{
+    public function up()
+    {
+        Schema::create('order_items', function (Blueprint $table) {
+            $table->id();
+            $table->foreignIdFor(Order::class);
+            $table->foreignIdFor( OrderItem::class, 'parent_id')->nullable();
+
+            // TODO - TBD - change to Type=>class mapping instead of Morph??  using morph forces the sellable to be a Model!
+            $table->morphs('sellable');
```

Comment:
This is probably the biggest TBD (something I hit with discounts and vouchers).  Using a morph **requires** that there is a DB model / record to morph to.  In the case of discounts, I want to be able to support different types of discount rules and class implementations, but I didn't want to force each of those implementations to require its own model / DB table.  So, I found using a "type" designation in place of a morph worked better.  I could then map the type => class and instantiate the class using the `Discount` model as a parameter.

It seems like it may be a similar decision here.  Is it better for every `Sellable` to always be backed by a model / table, or better to define the interface to support construction w/the cart item or order item?
```
interface Sellable {
public static function fromCartItem(CartItemInterface $cartItem): self;
public static function fromOrderItem(OrderItemInterface $orderItem): self;
...
}
```

Reply:
In general, it sounds like keeping Sellable as a model would be good for a start. Perhaps in the future in some v2 add `fromCartItem`?

Reply:
It makes sense for me to require every `Sellable` to always be backed by a model / table.